configuration.go

```diff
@@ -0,0 +1,121 @@
+package govh
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"strings"
+
+	"golang.org/x/sys/unix"
+	"gopkg.in/ini.v1"
+)
+
+// Use variables for easier test overload
+var (
+	systemConfigPath = "/etc/ovh.conf"
+	userConfigPath   = "/.ovh.conf" // prefixed with homeDir
+	localConfigPath  = "./ovh.conf"
+)
+
+// currentUserHome attempts to get current user's home directory
+func currentUserHome() (string, error) {
+	usr, err := user.Current()
+	if err != nil {
+		return "", err
+	}
+	return usr.HomeDir, nil
+}
+
+// appendConfigurationFile only if it exists. We need to do this baecause
+// ini package will fail to load configuration at all if a configuration
+// file is missing. This is racy, but better than always failing.
+func appendConfigurationFile(cfg *ini.File, path string) {
+	if unix.Access(path, unix.R_OK) == nil {
+		cfg.Append(path)
+	}
+}
+
+// loadConfig loads client configuration from params, environments or configuration
+// files (by order of decreasing precedence).
+//
+// loadConfig will check OVH_CONSUMER_KEY, OVH_APPLICATION_KEY, OVH_APPLICATION_SECRET
+// and OVH_ENDPOINT environment variables. If any is present, it will take precedence
+// over any configuration from file.
+//
+// Configuration files are ini files. They share the same format as python-ovh,
+// node-ovh, php-ovh and all other wrappers. If any wrapper is configured, all
+// can re-use the same configuration. loadConfig will check for configuration in:
+//
+// - ./ovh.conf
+// - $HOME/.ovh.conf
+// - /etc/ovh.conf
+//
+func (c *Client) loadConfig(endpointName string) error {
+	// Load configuration files by order of increasing priority. All configuration
+	// files are optional. Only load file from user home if home could be resolve
+	cfg := ini.Empty()
+	appendConfigurationFile(cfg, systemConfigPath)
+	if home, err := currentUserHome(); err == nil {
+		appendConfigurationFile(cfg, home+userConfigPath)
```

Comment:
It might be safer to use [filepath.Join](https://golang.org/pkg/path/filepath/#Join) to join paths